staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -881,8 +881,10 @@ func NewDefaultAPIGroupInfo(group string, scheme *runtime.Scheme, parameterCodec
 }
 
 // getOpenAPIModels is a private method for getting the OpenAPI models
-func (s *GenericAPIServer) getOpenAPIModels(apiPrefix string, apiGroupInfos ...*APIGroupInfo) (*spec.Swagger, error) {
-	if s.openAPIConfig == nil {
+func (s *GenericAPIServer) getOpenAPIModels(apiPrefix string, apiGroupInfos ...*APIGroupInfo) (map[string]*spec.Schema, error) {
+	if s.openAPIV3Config == nil {
+		//!TODO: A future work should add a requirement that
+		// OpenAPIV3 config is required. May require some refactoring of tests.
 		return nil, nil
 	}
 	pathsToIgnore := openapiutil.NewTrie(s.openAPIConfig.IgnorePrefixes)
```

Comment:
@alexzielenski,  This change is present on v1.27.

Isn't 
```
pathsToIgnore := openapiutil.NewTrie(s.openAPIConfig.IgnorePrefixes)
```

wrong if we expect `openAPIV3Config` to be present the also should read IgnorePrefixes from openAPIV3Config i.e.

```
pathsToIgnore := openapiutil.NewTrie(s.openAPIV3Config.IgnorePrefixes)
```

If you agree I can open PR for release branches.

Reply:
> @alexzielenski, This change is present on v1.27.

the revert also went into 1.27 (https://github.com/kubernetes/kubernetes/pull/116062)

Reply:
oh, I guess the revert was reverted as well - https://github.com/kubernetes/kubernetes/pull/116106

Reply:
This issue is fixed by https://github.com/kubernetes/kubernetes/pull/114998 which unfortunately had just missed the merge deadline and was merged into 1.28 and not 1.27. We had intended to back port this and there was one open that got closed. https://github.com/kubernetes/kubernetes/pull/119583. Ill re-open a cherry pick

Reply:
Thank you @alexzielenski.